include/deal.II/lac/linear_operator.h

```diff
@@ -727,68 +723,39 @@ identity_operator(const std::function<void(Range &, bool)> &reinit_vector)
 /**
  * @relates LinearOperator
  *
- * Returns a LinearOperator that is the null operator
- * from the vector space @p Domain to the vector space @p Range.
- *
- * The function takes two <code>std::function</code> objects @ref
- * reinit_range_vector and reinit_domain_vector as arguments to initialize the
- * <code>reinit_range_vector</code> and <code>reinit_domain_vector</code>
- * objects of the LinearOperator object.
+ * Returns a nulled variant of the LinearOperator @p op, i.e. with
+ * optimized LinearOperator::vmult, LinearOperator::vmult_add, etc.
+ * functions and with LinearOperator::is_null_operator set to true.
  *
  * @ingroup LAOperators
  */
-
-template <typename Domain,
-          typename Range>
-LinearOperator<Domain, Range>
-null_operator(const std::function<void(Domain &, bool)> &reinit_domain_vector,
-              const std::function<void(Range &,  bool)> &reinit_range_vector)
+template <typename Range, typename Domain>
+LinearOperator<Range, Domain>
+null_operator(const LinearOperator<Range, Domain> &op)
 {
-  LinearOperator<Domain, Range> return_op(/*is_null_operator = */true);
+  auto return_op = op;
 
-  return_op.reinit_range_vector  = reinit_range_vector;
-  return_op.reinit_domain_vector = reinit_domain_vector;
+  return_op.is_null_operator = true;
 
-  return_op.vmult = [](Range &v, const Domain &u)
+  return_op.vmult = [](Range &v, const Domain &)
   {
-    v = 0;
+    v = 0.;
```

Comment:
My experience with complex data types indicates that ultimately this will have to be changed to 
```
  v = typename Range::value_type();
```
But it can of course wait till we hit this problem the first time.